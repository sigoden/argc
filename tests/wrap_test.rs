macro_rules! snapshort_wrap {
    ($source:expr, $width:expr) => {
        let args: Vec<String> = vec!["prog --help".into()];
        let values = argc::eval(None, $source, &args, Some($width)).unwrap();
        let output = argc::ArgcValue::to_shell(values);
        insta::assert_snapshot!(output);
    };
}

#[test]
fn test_wrap() {
    let script: &str = r###"
# @option --foo Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Neque laoreet suspendisse libero id. 
# @arg target Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Sed viverra tellus in hac habitasse platea.
# @cmd Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Arcu cursus euismod quis viverra. 
foo() {
}
"###;
    snapshort_wrap!(script, 80);
}
