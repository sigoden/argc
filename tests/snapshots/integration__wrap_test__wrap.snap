---
source: tests/wrap_test.rs
expression: output
---
cat >&2 <<-'EOF' 
USAGE: prog --help [OPTIONS] [TARGET] <COMMAND>

ARGS:
  [TARGET]  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua. Sed
            viverra tellus in hac habitasse platea.

OPTIONS:
      --foo <FOO>  Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                   sed do eiusmod tempor incididunt ut labore et dolore magna
                   aliqua. Neque laoreet suspendisse libero id.
  -h, --help       Print help

COMMANDS:
  foo  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
       tempor incididunt ut labore et dolore magna aliqua. Arcu cursus euismod
       quis viverra.

EOF
exit 0
